File: src/main/java/chloe/movietalk/dto/response/user/UserInfo.java
package chloe.movietalk.dto.response.user;

import chloe.movietalk.domain.SiteUser;
import chloe.movietalk.domain.enums.UserRole;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Getter
@NoArgsConstructor
public class UserInfo {

    @Schema(description = "사용자 ID")
    private UUID id;

    @Schema(description = "이메일")
    private String email;

    @Schema(description = "닉네임")
    private String nickname;

    @Schema(description = "권한")
    private UserRole role;

    @Builder
    public UserInfo(UUID id, String email, String nickname, UserRole role) {
        this.id = id;
        this.email = email;
        this.nickname = nickname;
        this.role = role;
    }

    public static UserInfo fromEntity(SiteUser user) {
        return UserInfo.builder()
                .id(user.getId())
                .email(user.getEmail())
                .nickname(user.getNickname())
                .role(user.getRole())
                .build();
    }
}

File: src/main/java/chloe/movietalk/dto/response/user/UserInfo.kt
package chloe.movietalk.dto.response.user

import chloe.movietalk.domain.SiteUser
import chloe.movietalk.domain.enums.UserRole
import io.swagger.v3.oas.annotations.media.Schema
import java.util.*

data class UserInfo(
    @field:Schema(description = "사용자 ID")
    val id: UUID,
    
    @field:Schema(description = "이메일")
    val email: String,
    
    @field:Schema(description = "닉네임")
    val nickname: String,
    
    @field:Schema(description = "권한")
    val role: UserRole
) {
    companion object {
        @JvmStatic
        fun fromEntity(user: SiteUser): UserInfo {
            return UserInfo(
                id = requireNotNull(user.id) { "User ID must not be null"},
                email = user.email,
                nickname = user.nickname,
                role = user.role
            )
        }
    }
}